File: port_CSharp/grun_tokens.cs
using Antlr4.Runtime;
using System;

[assembly: CLSCompliant(false)]

namespace grun_tokens
{
    static class Program
    {
        static void Main(string[] args)
        {
            var input = CharStreams.fromPath(args[0]); // default command line argument for IDE: ..\..\..\test.py
            var lexer = new PythonLexer(input);
            var tokens = new CommonTokenStream(lexer);
            var parser = new PythonParser(tokens);

            tokens.Fill();
            foreach (IToken t in tokens.GetTokens())
            {
                Console.WriteLine(GetTokenMetaDataWithTokenName(parser, t));
            }

            parser.file_input();
        }

        private static string GetTokenMetaDataWithTokenName(PythonParser parser, IToken token)
        {
            String metaData = token.ToString()!;          // original format: [@TokenIndex,StartIndex:StopIndex='Text',<TokenType>,channel=Channel,Line:Column]
            int lesserPos = metaData.LastIndexOf(",<");
            int greaterPos = metaData.LastIndexOf(">,");
            return metaData.Substring(0, lesserPos + 2) // modified format:   [@TokenIndex,StartIndex:StopIndex='Text',<TokenName>,channel=Channel,Line:Column]
                 + parser.Vocabulary.GetSymbolicName(token.Type)
                 + metaData.Substring(greaterPos);

        }
    }
}

File: port_JavaScript/grun_tokens.js
import antlr4 from "antlr4";
import PythonLexer from "./PythonLexer.js";
import PythonParser from "./PythonParser.js";

function getTokenMetaDataWithTokenName(token) {
    const tokenText = token.text.replaceAll("\r", "\\r").replaceAll("\n", "\\n");
    const tokenName = token.type == -1 ? "EOF" : parser.symbolicNames[token.type];
    const optionalChannel = token.channel == 0 ? "" : "channel=" + token.channel + ",";

    //      [@tokenIndex,start:stop='tokenText',<tokenName>,channel=channel,line:column]
    return "[@" + token.tokenIndex + "," + token.start + ":" + token.stop + "='" + tokenText + "',<" + tokenName + ">," + optionalChannel + token.line + ":" + token.column + "]";
}

const input = antlr4.CharStreams.fromPathSync(process.argv[2]);
const lexer = new PythonLexer(input);
const tokens = new antlr4.CommonTokenStream(lexer);
const parser = new PythonParser(tokens);

tokens.fill();
for (const token of tokens.tokens) {
    console.log(getTokenMetaDataWithTokenName(token));
}
const tree = parser.file_input();
